### PR TITLE
comlplete main page component and connect other parts to api

### PR DESCRIPTION
**Pull request by Rahsaam**
please review my code , and i wanted watch movies and series according to their apis , give some point for this challenge 
its in movie-list component

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
+    
+      <navbar/>
+
     <main-page/>
   </div>
 </template>
 
 <script setup>
     import mainPage from './components/MoviePages/MainPage.vue'
+    import navbar from './components/dls/Navbar.vue'
```
Comment (amRashidi) [inline src/App.vue]: please use global directory address "@/components/dls/Navbar.vue"

File: src/components/Aside/ComingSoon.vue [hunk 2]
```diff
@@ -1,24 +1,19 @@
...
-            <h2 class="text-yellow-400">Comings</h2>
-            <ul class="flex gap-x-2">
-                <li><a href="">more</a></li>
-            </ul>
-        </div>
-        <div class="grid grid-cols-1 lg:grid-cols-4 md:grid-cols-5 sm:grid-cols-4 gap-4 mt-3 overflow-hidden">
-            <div v-for="movie in data" :key="movie.id" class="relative mx-auto w-full cursor-pointer" aria-selected="true">
-                <img :src="movie.img" alt="" class="w-full lg:h-full rounded-xl  h-28 object-cover">
+    
+            <div class="relative mx-auto w-full cursor-pointer" aria-selected="true">
```
Comment (amRashidi) [inline src/components/Aside/ComingSoon.vue:3]: why you use aria-selected?

File: src/components/Aside/TvShows.vue [hunk 3]
```diff
@@ -1,24 +1,19 @@
...
-                            <img :src="tvShow.img" alt="" class="w-full lg:h-full rounded-xl   h-28 object-cover">
-                            <div class="text-overlay text-center absolute inset-0 flex flex-col justify-center items-center  lg:w-full h-full   overflow-hidden bg-black bg-opacity-70 p-6 rounded-xl backdrop-blur-lg opacity-0 duration-300 text-xl hover:opacity-90">
-                                    <span class="text-sm text-white">{{tvShow.name}}</span>          
-                            </div>
-                        </div>
-                    </div>
-                </article>
+  
+    <div class="relative mx-auto w-full cursor-pointer" aria-selected="true">
+        <img :src="src" alt="" class="w-full lg:h-full rounded-xl   h-28 object-cover">
```
Comment (amRashidi) [inline src/components/Aside/TvShows.vue]: fill alt attribute with title

File: src/components/MainMovies/SeriesNader.vue [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+ rating: Number,
+ genereNames: Array
+})
+
+const year = computed(() => new Date(props.releaseDate).getFullYear())
+</script>
+
+<style>
+
+</style>
```
Comment (amRashidi) [inline src/components/MainMovies/SeriesNader.vue]: remove empty style tags

File: src/components/MoviePages/Login.vue [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    </div>
+</template>
+
+<script setup>
+
+</script>
+
+<style>
+.login-box {
+    background: url('../../assets/Movie-Wallpaper-HD-Free-download\ \(1\).jpg');
```
Comment (amRashidi) [inline src/components/MoviePages/Login.vue]: use global address format

File: src/components/MoviePages/MainPage.vue [hunk 6]
```diff
@@ -1,38 +1,59 @@
...
+  <middleNavbar/>
   <div class="md:h-32 h-16"></div>
   <section id="trend" class=" mx-auto mt-10 container max-w-sm sm:max-w-2xl md:max-w-5xl lg:max-w-7xl scroll-mt-16">
             <div class="flex justify-between">
                 <h2 class="text-yellow-400 text-2xl">Trend movies</h2>
                 <button class="btn-secodary">more...</button>
             </div>
             <div class="w-full overflow-x-scroll whitespace-no-wrap mx-auto mt-5">
-                <div class="flex flex-row space-x-3 mx-auto [&::-webkit-scrollbar]:hidden">
+                <div class="flex flex-row space-x-3 mx-auto [&::-webkit-scrollbar]:hidden" v-if="popularMovies.length">
```
Comment (amRashidi) [inline src/components/MoviePages/MainPage.vue:11]: what is this?
[&::-webkit-scrollbar]:hidden"

File: src/components/MoviePages/MainPage.vue [hunk 7]
```diff
@@ -1,38 +1,59 @@
...
                     :key="movie.id"
                     :rating="movie.vote_count"
                     :release-date="movie.release_date"
-                    :src="`${API_IMAGE_BASE_URL}${API_IMAGE_SIZE}${movie.poster_path}`"
+                    :src="`${API_IMAGE_BASE_URL}${API_IMAGE_SIZE_MD}${movie.poster_path}`"
                     :title="movie.original_title"
-                    :imdb="movie.vote_average"/>  
+                    :imdb="movie.vote_average"/> 
                 </div>
+                <div v-else>
```
Comment (amRashidi) [inline src/components/MoviePages/MainPage.vue:21]: by this scenario if we don't have data for movies we'll watch an infinite spinner.
is it currect?

File: src/components/MoviePages/MainPage.vue [hunk 8]
```diff
@@ -42,22 +63,148 @@ const options = {
   }
 };
 
+const fetchTrendMovies = async () => {
+  try {
+    await new Promise(resolve => setTimeout(resolve, 2000));
+    const response = await fetch('https://api.themoviedb.org/3/movie/popular?language=en-US&page=1', options);
```
Comment (amRashidi) [inline src/components/MoviePages/MainPage.vue]: use base url global variable

File: src/components/MoviePages/MainPage.vue [hunk 9]
```diff
@@ -42,22 +63,148 @@ const options = {
...
+    popularMovies.value = data.results;
+  } catch (err) {
+    console.error(err);
+  }
+};
+
+
+ const fetchMovies = async () => {
+  try {
+    await new Promise(resolve => setTimeout(resolve, 2000));
```
Comment (amRashidi) [inline src/components/MoviePages/MainPage.vue]: why?

File: src/components/MoviePages/MainPage.vue [hunk 10]
```diff
@@ -42,22 +63,148 @@ const options = {
...
+    const response = await fetch('https://api.themoviedb.org/3/trending/movie/day?language=en-US', options);
```
Comment (amRashidi) [inline src/components/MoviePages/MainPage.vue]: global base url

File: src/components/MoviePages/MainPage.vue [hunk 11]
```diff
@@ -42,22 +63,148 @@ const options = {
...
+    if (!response.ok) {
+      throw new Error('This URL does not exist');
+    }
+    const data = await response.json();
+    // mainMovies.value = data.results;
+
+    const genereResponse = await fetch('https://api.themoviedb.org/3/genre/movie/list?language=en', options)
```
Comment (amRashidi) [inline src/components/MoviePages/MainPage.vue]: same

File: src/components/MoviePages/MainPage.vue [hunk 12]
```diff
@@ -42,22 +63,148 @@ const options = {
...
+      genreMap[genre.id] = genre.name;
+    });
+    
+
+    // Assign genre names to each movie
+    const moviesWithGenres = data.results.map(movie => {
+      const genreNames = movie.genre_ids.map(genreId => genreMap[genreId]);
+      return {
+        ...movie,
+        genre_names: genreNames,
```
Comment (amRashidi) [inline src/components/MoviePages/MainPage.vue:95]: no need for this function...
you can do. this with a simple Map when you showing the component

File: src/components/MoviePages/MainPage.vue [hunk 13]
```diff
@@ -42,22 +63,148 @@ const options = {
...
+    
+  } catch (err) {
+    console.error(err);
+  }
+};
+
+
+const fetchUpcomings = async () => {
+  try {
+    const response = await fetch('https://api.themoviedb.org/3/tv/popular?language=en-US&page=1', options)
```
Comment (amRashidi) [inline src/components/MoviePages/MainPage.vue]: base url

File: src/components/MoviePages/MainPage.vue [hunk 14]
```diff
@@ -42,22 +63,148 @@ const options = {
...
+    // console.log(data);
+  } catch(err) {
+    console.log(err);
+  }
+};
+
+
+const fetchOldMovies = async () => {
+  try {
+    const response = await fetch('https://api.themoviedb.org/3/movie/top_rated?language=en-US&page=1', options)
```
Comment (amRashidi) [inline src/components/MoviePages/MainPage.vue]: same

File: src/components/MoviePages/MainPage.vue [hunk 15]
```diff
@@ -42,22 +63,148 @@ const options = {
...
+    oldMovies.value = data.results
+  } catch(err) {
+    console.log(err);
+  }
+}
+
+
+const fetchTvShows = async () => {
+  try {
+    const response = await fetch('https://api.themoviedb.org/3/trending/tv/day?language=en-US', options)
```
Comment (amRashidi) [inline src/components/MoviePages/MainPage.vue]: same

File: src/components/MoviePages/MainPage.vue [hunk 16]
```diff
@@ -42,22 +63,148 @@ const options = {
...
+  } catch(err) {
+    console.log(err);
+  }
+}
+
+
+const fetchSeries = async () => {
+  try {
+    await new Promise(resolve => setTimeout(resolve, 2000));
+    const response = await fetch('https://api.themoviedb.org/3/trending/all/day?language=en-US', options)
```
Comment (amRashidi) [inline src/components/MoviePages/MainPage.vue]: same

File: src/components/dls/Sppiner.vue [hunk 17]
```diff
@@ -0,0 +1,24 @@
+<template>
+    <div class="spin"></div>
+</template>
+  
+  <script setup>
```
Comment (amRashidi) [inline src/components/dls/Sppiner.vue]: remove empty script

File: src/components/dls/MovieList.vue [hunk 18]
```diff
@@ -41,13 +57,45 @@
...
+
+    const options = {
+        method: 'GET',
+        headers: {
+        accept: 'application/json',
+        Authorization: 'Bearer eyJhbGciOiJIUzI1NiJ9.eyJhdWQiOiI4YjZjMDBlZWVmYzdlZTk0OWFmNGU2MDcyNTU2ZjZhOCIsInN1YiI6IjY0YTJjZDFlOGUyMGM1MDEwZDRlMTc5MiIsInNjb3BlcyI6WyJhcGlfcmVhZCJdLCJ2ZXJzaW9uIjoxfQ.ZkW2-Er08cN1FMrSp1vVZIwbMwJc0QIyWQY1WxJrx7s'
+        }
+    };
+
+    // watch(() => updateData.value
```
Comment (amRashidi) [inline src/components/dls/MovieList.vue]: remove comments

File: src/components/dls/MovieList.vue [hunk 19]
```diff
@@ -41,13 +57,45 @@
...
+    const moviesData = ref([])
+    const seriesData = ref([])
 
-   
+
+    const options = {
+        method: 'GET',
+        headers: {
+        accept: 'application/json',
+        Authorization: 'Bearer eyJhbGciOiJIUzI1NiJ9.eyJhdWQiOiI4YjZjMDBlZWVmYzdlZTk0OWFmNGU2MDcyNTU2ZjZhOCIsInN1YiI6IjY0YTJjZDFlOGUyMGM1MDEwZDRlMTc5MiIsInNjb3BlcyI6WyJhcGlfcmVhZCJdLCJ2ZXJzaW9uIjoxfQ.ZkW2-Er08cN1FMrSp1vVZIwbMwJc0QIyWQY1WxJrx7s'
```
Comment (amRashidi) [inline src/components/dls/MovieList.vue:73]: avoid hardcoding token like this.
use a variable.

**Review by amRashidi — CHANGES_REQUESTED** [review]
caution: please edit all off the import url path and convert the import model to global import (@/)

File: src/components/Aside/ComingSoon.vue (hunk 2 above)
Comment (Rahsaam) [reply src/components/Aside/ComingSoon.vue:3]: in my opinion it helps users with disabilities can understand the current selection

**Pull request merged**